src/GitHubExtension/DeveloperId/DeveloperId.cs

```diff
@@ -55,15 +55,16 @@ public Windows.Security.Credentials.PasswordCredential GetCredential(bool refres
             return RefreshDeveloperId();
         }
 
-        return CredentialVault.GetCredentialFromLocker(LoginId);
+        var credential = DeveloperIdProvider.GetInstance().GetCredentials(this) ?? throw new InvalidOperationException("Invalid credential present for valid DeveloperId");
+        return credential;
     }
 
     public Windows.Security.Credentials.PasswordCredential RefreshDeveloperId()
     {
         // Setting to MaxValue, since GitHub doesn't forcibly expire tokens currently.
         CredentialExpiryTime = DateTime.MaxValue;
         DeveloperIdProvider.GetInstance().RefreshDeveloperId(this);
-        var credential = CredentialVault.GetCredentialFromLocker(LoginId);
+        var credential = DeveloperIdProvider.GetInstance().GetCredentials(this) ?? throw new InvalidOperationException("Invalid credential present for valid DeveloperId");

```

Comment:
In the future, yes. These should be defined in the SDK so that DevHome and extensions are on the same page about exception handling. Otherwise, custom exception objects are shown as their base class exception on DevHome side. Will consider adding custom exceptions in a future PR once error status feature on DevHome UI is in a less fluid state. 
Also, RefreshDeveloperId() isn't used right now since GitHub OAuth tokens don't support refresh flow.